src/functions/validateTime.js

```diff
@@ -30,10 +54,12 @@ export const validateTime = (bookings, date, start, end) => {
     alert("You can't book more than 60 days in advance.");
     return false;
   } else {
-    for (let i = 0; i < bookings.length; ++i) {
+    const dissectedBookings = dissectBookings([...bookings]);
+
+    for (let i = 0; i < dissectedBookings.length; ++i) {
       if (
-        moment(bookings[i].start).isSame(startTime, 'hour') ||
-        moment(bookings[i].end).isSame(endTime, 'hour')
+        moment(dissectedBookings[i].startTime).isSame(startTime, 'hour') ||
+        moment(dissectedBookings[i].endTime).isSame(endTime, 'hour')
```

Comment:
algorithm seems a bit wrong

when `startTime` is `8am` and `endTime` is 12pm`, and there's an existing booking with `9am` to `11am`, 
```
moment(dissectedBookings[i].startTime).isSame(startTime, 'hour') ||
  moment(bookings[i].end).isSame(endTime, 'hour')
```
seems to always be false, which results in booking being valid when it actually isn't.

Reply:
> algorithm seems a bit wrong
> 
> when `startTime` is `8am` and `endTime` is `12pm`, and there's an existing booking with `9am` to `11am`,
> 
> ```
> moment(dissectedBookings[i].startTime).isSame(startTime, 'hour') ||
>   moment(bookings[i].end).isSame(endTime, 'hour')
> ```
> 
> seems to always be false, which results in booking being valid when it actually isn't.

Ohh I see, I kept in mind that max number of hours is 2 when doing that. Will commit a change soon, thanks for the heads up!